bindings/nodejs/src/account/storage/stronghold.rs

```diff
@@ -0,0 +1,202 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use futures::executor;
+use identity::account::DIDLease;
+use identity::account::Storage;
+use identity::account::Stronghold;
+use identity::core::decode_b58;
+use identity::core::encode_b58;
+use identity::crypto::PrivateKey;
+use identity::crypto::PublicKey;
+use napi::bindgen_prelude::Error;
+use napi::Result;
+
+use crate::account::NapiChainState;
+use crate::account::NapiDIDLease;
+use crate::account::NapiIdentityState;
+use crate::account::NapiKeyLocation;
+use crate::account::NapiSignature;
+use crate::did::NapiDID;
+use crate::error::NapiResult;
+
+#[napi]
+#[derive(Debug)]
+pub struct NapiStronghold(pub(crate) Stronghold);
+
+#[napi]
+impl NapiStronghold {
+  /// Creates an instance of `Stronghold`.
+  /// Prefer to use Stronghold.new(snapshot, password, dropsave)
+  #[napi(factory)]
+  pub fn create(snapshot: String, password: String, dropsave: Option<bool>) -> NapiStronghold {
+    let stronghold_result = executor::block_on(NapiStronghold::new(snapshot, password, dropsave));
+    match stronghold_result {
+      Ok(stronghold) => stronghold,
+      Err(e) => panic!("Unable to create Stronghold: {:?}", e),
+    }
+  }
+
+  /// Creates an instance of `Stronghold`.
+  #[napi]
+  pub async fn new(
+    snapshot: String,
+    password: String,
+    dropsave: Option<bool>,
+  ) -> Result<NapiStronghold> {
+    Ok(NapiStronghold(
+      Stronghold::new(&snapshot, &*password, dropsave)
+        .await
+        .napi_result()?,
+    ))
+  }
+
+  /// Returns whether save-on-drop is enabled.
+  #[napi(getter)]
+  pub fn dropsave(&self) -> bool {
+    self.0.dropsave()
+  }
+
+  /// Set whether to save the storage changes on drop.
+  /// Default: true
+  #[napi(setter)]
+  pub fn set_dropsave(&mut self, dropsave: bool) {
+    self.0.set_dropsave(dropsave);
+  }
+
+  /// Sets the account password.
+  #[napi]
+  pub async fn set_password(&self, password: Vec<u32>) -> Result<()> {
```

Comment:
Unfortunately  we can't. 
```
#[napi] has type `napi::bindgen_prelude::Uint8Array` which is not `Send`
```

Reply:
Looks like a shortcoming of napi. It requires the `Send` bound on `async fn`s because of `tokio`. Guess `Uint8Array` can only be used in sync functions for now. As a slight improvement, you could put the `Vec<u32>` -> `Vec<u8>` conversion in a separate function and call it from `key_sign` and `set_password`.